### PR TITLE
JER in DataFormat

### PR DESCRIPTION
**Pull request by jhgoh**
First attempt to implement #221 
JER is moved to DataFormat definition.

Adding fancy search algorithm as a bonus :-)

### CODE REVIEW

File: DataFormats/interface/Jet.h [hunk 1]
```diff
@@ -75,15 +75,12 @@ namespace cat {
...
-    void setSmearedResUp(float f) { smearedResUp_ = f;}
 
     float shiftedEnDown() const {return  shiftedEnDown_;}
     float shiftedEnUp() const  {return  shiftedEnUp_;}
-    float smearedRes() const {return  smearedRes_;}
-    float smearedResDown() const {return  smearedResDown_;}
-    float smearedResUp() const {return  smearedResUp_;}
+    float smearedRes(int direction=0) const; // 0, +1, -1 for smeared, smearedUp, smearedDown
+    float smearedResUp() const { return smearedRes(+1); };
+    float smearedResDown() const { return smearedRes(-1); };
```
Comment (jshlee) [inline DataFormats/interface/Jet.h:84]: can u remove line 116-118
smearedResX member variables

Comment (jhgoh) [inline DataFormats/interface/Jet.h:84]: Done.

**Comment by jshlee** [comment]
+1

**Comment by jhgoh** [comment]
And some helper functions are added to calculate MET uncertainty variations.
This part is not ready to use for end-users nor fully tested - review needed.

**Pull request merged**